ПЕ-22б Томашов М.: Создать таблицу TrainingPlan.lesson2lecturer

maxtom93 opened the pull request.

maxtom93 committed:
ПЕ-22б Томашов М.: Создать таблицу TrainingPlan.lesson2lecturer

dha01 commented on SQL/Task_40/create table TrainingPlan.lesson2lecturer.sql:
```diff
@@ -0,0 +1,12 @@
+-- �������� ����� TrainingPlan
```
Неправильная кодировка файла

dha01 commented on SQL/Task_40/create table TrainingPlan.lesson2lecturer.sql:
```diff
@@ -0,0 +1,12 @@
+	lesson2lecturer int identity,
+	lesson int not null,
+	person int not null,
+	constraint PK__lesson primary key (lesson),
```
У каждой таблицы должно специальное быть поле - идентификатор с названием как у таблицы (lesson2lecturer ) оно и должно быть первичным ключем таблицы

maxtom93 committed:
ПЕ-22б: Томашов М. Исправил:

У каждой таблицы должно специальное быть поле - идентификатор с названием как у таблицы (lesson2lecturer ) оно и должно быть первичным ключем таблицы

dha01 commented on SQL/Task_40/create table TrainingPlan.lesson2lecturer.sql:
```diff
@@ -0,0 +1,13 @@
+	person int not null,
+	constraint PK__lesson2lecturer primary key (lesson2lecturer),
+	constraint FK__lesson2lecturer__person foreign key (person) references Person.person,
+	constraint FK__lesson2lecturer__person foreign key (lesson) references TraningPlan.lesson
```
Имя FK не соответствует полю на которое ссылается
FK__lesson2lecturer__lesson
+Немного теории:
в БД не может существовать 2-х объектов одинаковыми именами (FK тоже является объектом в бд как таблица, схема и д.р.).

maxtom93 committed:
ПЕ-22б: Томашов М.В. Исправил
